src/util/kdtree.jl

```diff
@@ -16,14 +16,16 @@ not shapes of nonzero size.)
 """
 mutable struct KDTree{K,S<:Shape{K}}
     s::Vector{S}
+    s_index::Vector{Int}
```

Comment:
Why do we need to store an array of indices?

Reply:
I guess you need to store where the structures were in the original array.

Reply:
Yep exactly.